js/components/lab-model.js

```diff
@@ -20,8 +20,10 @@ export default class LabModel extends PureComponent {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.model !== prevProps.model) {
-      this.setState({loading: true});
+    if (this.props.model !== prevProps.model && !this.props.liveData) {
+      this.setState({ loading: true });
+    } else {
+      this.setState({ loading: false });
```

Comment:
This one was a strange one - it only became an issue if I switched model types after having connected to the sensor.  I can't reproduce the issue on the non-sensor version.